diff --git a/src/data_manager/pos_list_provider.cc b/src/data_manager/pos_list_provider.cc
index 5b743f9a1..5c31eea2a 100644
--- a/src/data_manager/pos_list_provider.cc
+++ b/src/data_manager/pos_list_provider.cc
@@ -51,25 +51,16 @@ namespace {
 
 }  // namespace
 
-PosListProvider::PosListProvider() {
+std::vector<std::string> PosListProvider::GetPosList() const {
+
   SerializedStringArray array;
   CHECK(array.Init(LoadEmbeddedFile(kPosArray)));
 
-  std::vector<std::string> pos_list_(array.size());
+  std::vector<std::string> pos_list(array.size());
   for (absl::string_view pos : array) {
-    if (pos == "名詞") {
-      pos_list_default_index_ = pos_list_.size();
-    }
-    pos_list_.push_back({pos.data(), pos.size()});
+    pos_list.push_back({pos.data(), pos.size()});
   }
-}
-
-std::vector<std::string> PosListProvider::GetPosList() const {
-  return pos_list_;
-}
-
-int PosListProvider::GetPosListDefaultIndex() const {
-  return pos_list_default_index_;
+  return pos_list;
 }
 
 }  // namespace mozc
diff --git a/src/data_manager/pos_list_provider.h b/src/data_manager/pos_list_provider.h
index 76250b208..6b61be826 100644
--- a/src/data_manager/pos_list_provider.h
+++ b/src/data_manager/pos_list_provider.h
@@ -39,14 +39,7 @@ namespace mozc {
 
 class PosListProvider : public PosListProviderInterface {
  public:
-  PosListProvider();
-
   std::vector<std::string> GetPosList() const override;
-  int GetPosListDefaultIndex() const override;
-
- private:
-  std::vector<std::string> pos_list_;
-  int pos_list_default_index_ = 0;
 };
 
 }  // namespace mozc
diff --git a/src/dictionary/user_dictionary_test.cc b/src/dictionary/user_dictionary_test.cc
index 2c0a1fd1d..a3eb6c4bc 100644
--- a/src/dictionary/user_dictionary_test.cc
+++ b/src/dictionary/user_dictionary_test.cc
@@ -173,10 +173,7 @@ class UserPosMock : public UserPosInterface {
   }
 
   std::vector<std::string> GetPosList() const override {
-    return {{kNoun.data(), kNoun.size()}};
-  }
-  int GetPosListDefaultIndex() const override {
-    return 0;
+    return {};
   }
 
   bool GetPosIds(absl::string_view pos, uint16_t *id) const override {
diff --git a/src/dictionary/user_pos.cc b/src/dictionary/user_pos.cc
index 99eb85f50..de098c947 100644
--- a/src/dictionary/user_pos.cc
+++ b/src/dictionary/user_pos.cc
@@ -55,22 +55,19 @@ UserPos::UserPos(absl::string_view token_array_data,
   DCHECK_EQ(token_array_data.size() % 8, 0);
   DCHECK(SerializedStringArray::VerifyData(string_array_data));
   string_array_.Set(string_array_data);
-  InitPosList();
 }
 
-void UserPos::InitPosList() {
+std::vector<std::string> UserPos::GetPosList() const {
+  std::vector<std::string> pos_list;
   absl::flat_hash_set<uint16_t> seen;
   for (auto iter = begin(); iter != end(); ++iter) {
     if (!seen.insert(iter.pos_index()).second) {
       continue;
     }
     const absl::string_view pos = string_array_[iter.pos_index()];
-    if (pos == "名詞") {
-      // "名詞" is the default POS.
-      pos_list_default_index_ = pos_list_.size();
-    }
-    pos_list_.push_back({pos.data(), pos.size()});
+    pos_list.emplace_back(pos.data(), pos.size());
   }
+  return pos_list;
 }
 
 bool UserPos::IsValidPos(absl::string_view pos) const {
diff --git a/src/dictionary/user_pos.h b/src/dictionary/user_pos.h
index 1366c615b..fd68f871d 100644
--- a/src/dictionary/user_pos.h
+++ b/src/dictionary/user_pos.h
@@ -185,10 +185,7 @@ class UserPos : public UserPosInterface {
   UserPos &operator=(const UserPos &) = delete;
 
   // Implementation of UserPosInterface.
-  std::vector<std::string> GetPosList() const override { return pos_list_; }
-  int GetPosListDefaultIndex() const override {
-    return pos_list_default_index_;
-  }
+  std::vector<std::string> GetPosList() const override;
   bool IsValidPos(absl::string_view pos) const override;
   bool GetPosIds(absl::string_view pos, uint16_t *id) const override;
   bool GetTokens(absl::string_view key, absl::string_view value,
@@ -201,12 +198,8 @@ class UserPos : public UserPosInterface {
   }
 
  private:
-  void InitPosList();
-
   absl::string_view token_array_data_;
   SerializedStringArray string_array_;
-  std::vector<std::string> pos_list_;
-  int pos_list_default_index_ = 0;
 };
 
 }  // namespace dictionary
diff --git a/src/dictionary/user_pos_interface.h b/src/dictionary/user_pos_interface.h
index befa51d9b..d9f39a2c3 100644
--- a/src/dictionary/user_pos_interface.h
+++ b/src/dictionary/user_pos_interface.h
@@ -49,9 +49,6 @@ class PosListProviderInterface {
 
   // Gets possible list of POS which Mozc can handle.
   virtual std::vector<std::string> GetPosList() const = 0;
-
-  // Returns the index of the default POS in GetPosList.
-  virtual int GetPosListDefaultIndex() const = 0;
 };
 
 // Interface of the helper class used by POS.
diff --git a/src/dictionary/user_pos_test.cc b/src/dictionary/user_pos_test.cc
index 8b51b7b16..f4e75e564 100644
--- a/src/dictionary/user_pos_test.cc
+++ b/src/dictionary/user_pos_test.cc
@@ -62,11 +62,6 @@ class UserPosTest : public ::testing::Test {
   const testing::MockDataManager mock_data_manager_;
 };
 
-TEST_F(UserPosTest, GetPosListDefaultIndex) {
-  const std::vector<std::string> pos_list = user_pos_->GetPosList();
-  EXPECT_EQ(pos_list[user_pos_->GetPosListDefaultIndex()], "名詞");
-}
-
 TEST_F(UserPosTest, UserPosBasicTest) {
   const std::vector<std::string> pos_list = user_pos_->GetPosList();
   EXPECT_FALSE(pos_list.empty());
diff --git a/src/gui/dictionary_tool/dictionary_tool.cc b/src/gui/dictionary_tool/dictionary_tool.cc
index c90e0b208..0cb9451ec 100644
--- a/src/gui/dictionary_tool/dictionary_tool.cc
+++ b/src/gui/dictionary_tool/dictionary_tool.cc
@@ -423,9 +423,10 @@ DictionaryTool::DictionaryTool(QWidget *parent)
   delegate->SetItemList(pos_list);
   dic_content_->setItemDelegateForColumn(2, delegate);
 
-  // Set the default POS. It should be "名詞".
-  default_pos_ = pos_list[pos_list_provider_->GetPosListDefaultIndex()];
-  DCHECK(default_pos_ == "名詞") << "The default POS is not 名詞";
+  // Set the default POS to "名詞" indexed with 1.
+  constexpr absl::string_view kNoun = "名詞";
+  DCHECK(pos_list[1] == kNoun.data()) << "pos_list[1] is not " << kNoun;
+  default_pos_ = kNoun.data();
 
   // Set up the main table widget for dictionary contents.
   dic_content_->setColumnCount(4);
diff --git a/src/gui/word_register_dialog/word_register_dialog.cc b/src/gui/word_register_dialog/word_register_dialog.cc
index ad80ed33c..ca50cce9c 100644
--- a/src/gui/word_register_dialog/word_register_dialog.cc
+++ b/src/gui/word_register_dialog/word_register_dialog.cc
@@ -164,10 +164,8 @@ WordRegisterDialog::WordRegisterDialog()
     PartOfSpeechcomboBox->addItem(QString::fromUtf8(pos.c_str()));
   }
   // Set the default POS to "名詞" indexed with 1.
-  PartOfSpeechcomboBox->setCurrentIndex(
-      pos_list_provider_->GetPosListDefaultIndex());
-  DCHECK(PartOfSpeechcomboBox->currentText() == "名詞")
-      << "The default POS is not 名詞";
+  PartOfSpeechcomboBox->setCurrentIndex(1);
+  DCHECK(PartOfSpeechcomboBox->currentText() == "名詞") << "POS is not 名詞";
 
   // Create new dictionary if empty
   if (!session_->mutable_storage()->Exists().ok() ||
